modules/mcagar/docs/bdd.md

```diff
@@ -0,0 +1,224 @@
+# BDD testing
+
+The following are the primary modules used when testing TinyMCE using the behaviour driven development tests.
+
+## TinyHooks
+
+The `TinyHooks` module registers `before` and `after` hooks to set up TinyMCE in a test. It takes the TinyMCE settings and will then initialize the editor during the test suite setup phase. Additional module setup functions (such as a plugin constructor) can also be provided and will be called before the editor is initialized. The editor instance can be accessed by calling the `editor()` function on the object returned by the hook function.
+
+> 💡 **Note:** The editor is only initialized once per suite instead of per test, so be sure to close open dialogs, or reset any state after each test.
+
+Available functions:
+- `bddSetup(settings: RawEditorSettings, setupModules?: Array<() => void>, focusOnInit?: boolean)`
+  - Sets up the editor using the settings provided. This is the most common way to setup and test TinyMCE.
+- `bddSetupLight(settings: RawEditorSettings, setupModules?: Array<() => void>, focusOnInit?: boolean)`
+  - Similar to `bddSetup`, except that by default the editor `menubar`, `toolbar` and `statusbar` will be disabled. This helps to speed up tests where the UI isn't required since there's less to be rendered.
+- `bddSetupFromElement(settings: RawEditorSettings, setupElement: () => SetupElement, setupModules?: Array<() => void>, focusOnInit?: boolean)`
```

Comment:
Linking it isn't going to really work since it's in a code block. I definitely don't want to replicate it, as it'll likely become out of sync. Honestly, I don't think it's an issue looking at the actual type, as you already need to do that for things like `RawEditorSettings`.

Reply:
I should add the key point of the docs is to show what's available and where it might be best used.